src/snowflake/snowpark/dataframe.py

```diff
@@ -4367,6 +4367,7 @@ def _show_string(
                 self.limit(n, _emit_ast=False)._plan, **kwargs
             )
         else:
+            kwargs["show_limit"] = n
             res, meta = self._session._conn.get_result_and_metadata(
                 self._plan, **kwargs
```

Comment:
```suggestion
            res, meta = self._session._conn.get_result_and_metadata(
                self._plan, show_limit=n, **kwargs
```
why not just do this?